File: src/main/scala-2.11/com/frenchcoder/scalamones/ui/EventHandlerProperty.scala
package com.frenchcoder.scalamones.ui

import javafx.beans.property.ObjectPropertyBase
import javafx.event.{EventHandler, Event, EventType}

class EventHandlerProperty(val producer: AnyRef, val event: EventType[Event], val eventName: String, val setEventHandler: (EventType[Event], EventHandler[Event]) => Unit) extends ObjectPropertyBase[EventHandler[Event]] {
  protected override def invalidated = setEventHandler (event, get )
  def getBean = producer
  def getName = eventName
}

File: src/main/java/com/frenchcoder/scalamones/java/ui/EventHandlerProperty.java
package com.frenchcoder.scalamones.java.ui;

import javafx.beans.property.ObjectPropertyBase;
import javafx.event.Event;
import javafx.event.EventHandler;

/**
 *
 */
public class EventHandlerProperty extends ObjectPropertyBase<EventHandler<Event>> {

    private final EventProducerInterface eventProducer;
    private final String eventName;

    public EventHandlerProperty(EventProducerInterface eventProducer, String eventName) {
        this.eventProducer = eventProducer;
        this.eventName = eventName;
    }

    protected void invalidated() {
        eventProducer.setBasicEventHandler(WidgetPane.CLOSE_REQUEST_EVENT, (EventHandler) get());
    }

    public Object getBean() {
        return eventProducer;
    }

    public String getName() {
        return eventName;
    }
}
